src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let stateCopy = { ...state };
+  const resultingdArray = [];
+
+  for (const { type,
+    extraData: propertiesToAdd,
+    keysToRemove: propertiesToDelete } of actions) {
+    stateCopy = { ...stateCopy };
+
+    switch (type) {
+      case 'addProperties':
+        Object.assign(stateCopy, propertiesToAdd);
+        break;
+
+      case 'removeProperties':
+        for (const toRemove of propertiesToDelete) {
+          delete stateCopy[toRemove];
+        }
+        break;
+
+      case 'clear':
+        for (const key in stateCopy) {
+          delete stateCopy[key];
+        }
```

Comment:
Maybe you can just write stateCopy = {}; In this case, you don't need to use loop and use more memory.